pkg/plugins/policies/meshhttproute/api/v1alpha1/compare.go

```diff
@@ -0,0 +1,150 @@
+package v1alpha1
+
+import (
+	"slices"
+
+	common_api "github.com/kumahq/kuma/api/common/v1alpha1"
+	"github.com/kumahq/kuma/pkg/util/pointer"
+)
+
+func comparePath(a *PathMatch, b *PathMatch) int {
+	switch {
+	case a != nil && b == nil:
+		return -1
+	case a == nil && b != nil:
+		return 1
+	case a == nil && b == nil:
+		return 0
+	}
+
+	switch {
+	case a.Type == Exact && b.Type == Exact:
```

Comment:
Shouldn't it return 0 in this case?

Reply:
Would this be a bit simpler to rewrite this this way:

```go
switch{
case a.Type == b.Type:
   // imperfect syntax
   return cm.Compare(a.Value, b.Value)
case a.Type == Exact:
   return -1
case a.Type == PathPrefix:
  if b.Type == Exact { return 1 } else { return -1}
case a.Type == RegularExpression:
  return 1
}
```

In plain text:
- If types is the same order by length (for exact it's not required but at least we have an absolute order).
- If a.Type lower priority than b.Type then a.Type is priority (and vice versa)

We order the cases by priority so we don't have to do all types always.

Reply:
Why should it return 0? What if one has a method match and the other doesn't? I find that switch statement pretty hard to read, to be honest. We don't compare length for exact matches, just prefix matches.

Reply:
> If types is the same order by length (for exact it's not required but at least we have an absolute order).

We actually need them not to be reordered because we need stability when sorting.

Reply:
I fixed this but made it more explicit because I prefer not to have an implicit "RegularExpression is the 'last' kind" assumption built in.